subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JavaCompileSpec.java

```diff
@@ -18,16 +18,35 @@
 
 import org.gradle.api.internal.tasks.compile.processing.AnnotationProcessorDeclaration;
 
+import javax.annotation.Nullable;
 import java.io.File;
 import java.util.List;
 import java.util.Set;
 
 public interface JavaCompileSpec extends JvmLanguageCompileSpec {
+    enum JavaClassCompileOrder {
+        REVERSE_ALPHABETICALLY,
+        UNORDERED;
+
+        public static JavaClassCompileOrder fromInternalSystemProperty() {
+            return "reverse".equals(System.getProperty("org.gradle.internal.javac.class.compile.order")) ? REVERSE_ALPHABETICALLY : UNORDERED;
```

Comment:
❌ It seems like this property is only important for the tests. Is there a way to get rid of it? In a sense, the tests don't run the same code a user would run in their build.

If you leave this property in, then you should add some Javadoc on this enum.

Reply:
Yes, this is only important for the tests. I am using this flag since I need stable order of `.java` files passed to javac. And with stable order of `.java` files I can test, that we remove all classes that were written to a file system before a compilation failure.

Do you have any suggestion on how I could otherwise make an order of files passed to javac stable?

Reply:
Can you adjust the tests, so they don't rely on a stable order of files passed to Javac?

Reply:
So to be more clear, I don't know if I can test that any other way.

Reply:
Tbh, I don't like it either.

Reply:
If you have a few more files and the file with the compile error "in the middle" (aka lexicographically and by creation/modification date), then I suppose it will test the right things most of the time. WDYT?

Reply:
I can try :)

Reply:
I changed this, lets see how it works.

Reply:
I reverted back and I use stable order again for these tests. Without them testing this is very hard. I added Javadoc and limited the use of this stable order only for tests that require it.